// vue-tradingview-widgets/dist/tradingview-vue.js
import { defineComponent as a, ref as u, onMounted as v, openBlock as s, createElementBlock as d } from "vue";
window.TradingView = window.TradingView || {};
const _ = a({
  name: "WChart",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = u("tradingview-chart"), n = u("tradingview-chart-script"), i = {
      width: 980,
      height: 610,
      symbol: "NASDAQ:AAPL",
      interval: "D",
      timezone: "Etc/UTC",
      theme: "light",
      style: "1",
      locale: "en",
      toolbar_bg: "#f1f3f6",
      enable_publishing: !1,
      allow_symbol_change: !0,
      container_id: "tradingview-chart"
    }, o = () => typeof window < "u" && window.document && window.document.createElement, r = () => document.getElementById(n.value), g = () => r() !== null, h = (m) => {
      if (!o() || g())
        return;
      const p = document.createElement("script");
      p.id = n.value + Date.now().toString(36), p.type = "text/javascript", p.async = !0, p.src = "https://s3.tradingview.com/tv.js", p.onload = m, document.getElementsByTagName("head")[0].appendChild(p);
    }, f = () => {
      setTimeout(() => {
        if (typeof window.TradingView > "u")
          return;
        const m = { ...i, ...e.options };
        if (m.container_id !== t.value) {
          console.error('container_id in Chart component must be "tradingview-chart"');
          return;
        }
        new window.TradingView.widget(Object.assign({ container_id: t }, m));
      }, 300);
    };
    return v(() => {
      h(() => f());
    }), { container: t };
  }
}), c = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [i, o] of t)
    n[i] = o;
  return n;
}, b = ["id"];
function $(e, t, n, i, o, r) {
  return s(), d("div", { id: e.container }, null, 8, b);
}
const Ce = /* @__PURE__ */ c(_, [["render", $]]), l = (e, t, n, i) => {
  const o = u(t), r = u(n), g = u(), h = () => typeof window < "u" && window.document && window.document.createElement, f = () => document.getElementById(r.value), m = () => f() !== null, p = () => {
    if (!h() || m())
      return;
    const w = document.createElement("script");
    w.id = r.value, w.type = "text/javascript", w.async = !0, w.src = i, w.textContent = JSON.stringify(e), g.value && g.value.appendChild(w);
  };
  return v(() => {
    setTimeout(() => {
      p();
    }, 300);
  }), { container: o, tradingview: g };
}, y = a({
  name: "WCompanyProfile",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbol: "NASDAQ:AAPL",
      width: 480,
      height: 650,
      colorTheme: "light",
      isTransparent: !1,
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-company-profile",
      "tradingview-company-profile-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-symbol-profile.js"
    );
    return { container: n, tradingview: i };
  }
}), C = ["id"];
function T(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, C);
}
const Te = /* @__PURE__ */ c(y, [["render", T]]), S = a({
  name: "WCryptoMarket",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      width: "100%",
      height: "100%",
      defaultColumn: "overview",
      screener_type: "crypto_mkt",
      displayCurrency: "USD",
      colorTheme: "light",
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-crypto-market",
      "tradingview-crypto-market-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-screener.js"
    );
    return { container: n, tradingview: i };
  }
}), E = ["id"];
function F(e, t, n, i, o, r) {
  return s(), d("div", {
    id: e.container,
    ref: "tradingview"
  }, null, 8, E);
}
const Se = /* @__PURE__ */ c(S, [["render", F]]), U = a({
  name: "WEconomicCalendar",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      colorTheme: "light",
      isTransparent: !1,
      width: "510",
      height: "600",
      locale: "en",
      importanceFilter: "-1,0,1",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-economic-calendar",
      "tradingview-economic-calendar-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-events.js"
    );
    return { container: n, tradingview: i };
  }
}), D = ["id"];
function O(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, D);
}
const Ee = /* @__PURE__ */ c(U, [["render", O]]), N = a({
  name: "WForexCrossRates",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbol: "NASDAQ:AAPL",
      color: "blak",
      isTransparent: !1,
      largeChartUrl: "",
      displayMode: "regular",
      width: "100%",
      height: "100%",
      locale: "br",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-forex-cross-rates",
      "tradingview-forex-cross-rates-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-forex-cross-rates.js"
    );
    return { container: n, tradingview: i };
  }
}), M = ["id"];
function B(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, M);
}
const Fe = /* @__PURE__ */ c(N, [["render", B]]), X = a({
  name: "WForexHeatMap",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      width: 770,
      height: 400,
      currencies: ["EUR", "USD", "JPY", "GBP", "CHF", "AUD", "CAD", "NZD", "CNY"],
      isTransparent: !1,
      colorTheme: "light",
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-forex-heat-map",
      "tradingview-forex-heat-map-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-forex-heat-map.js"
    );
    return { container: n, tradingview: i };
  }
}), k = ["id"];
function A(e, t, n, i, o, r) {
  return s(), d("div", {
    id: e.container,
    ref: "tradingview"
  }, null, 8, k);
}
const Ue = /* @__PURE__ */ c(X, [["render", A]]), x = a({
  name: "WFundamentalData",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbol: "NASDAQ:AAPL",
      colorTheme: "light",
      isTransparent: !1,
      largeChartUrl: "",
      displayMode: "regular",
      width: 480,
      height: 830,
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-fundamental-data",
      "tradingview-fundamental-data-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-financials.js"
    );
    return { container: n, tradingview: i };
  }
}), j = ["id"];
function L(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, j);
}
const De = /* @__PURE__ */ c(x, [["render", L]]), P = a({
  name: "WMarketData",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      width: 770,
      height: 450,
      symbolsGroups: [
        {
          name: "Indices",
          originalName: "Indices",
          symbols: [
            {
              name: "FOREXCOM:SPXUSD",
              displayName: "S&P 500"
            },
            {
              name: "FOREXCOM:NSXUSD",
              displayName: "US 100"
            },
            {
              name: "FOREXCOM:DJI",
              displayName: "Dow 30"
            },
            {
              name: "INDEX:NKY",
              displayName: "Nikkei 225"
            },
            {
              name: "INDEX:DEU40",
              displayName: "DAX Index"
            },
            {
              name: "FOREXCOM:UKXGBP",
              displayName: "UK 100"
            }
          ]
        },
        {
          name: "Futures",
          originalName: "Futures",
          symbols: [
            {
              name: "CME_MINI:ES1!",
              displayName: "S&P 500"
            },
            {
              name: "CME:6E1!",
              displayName: "Euro"
            },
            {
              name: "COMEX:GC1!",
              displayName: "Gold"
            },
            {
              name: "NYMEX:CL1!",
              displayName: "Crude Oil"
            },
            {
              name: "NYMEX:NG1!",
              displayName: "Natural Gas"
            },
            {
              name: "CBOT:ZC1!",
              displayName: "Corn"
            }
          ]
        },
        {
          name: "Bonds",
          originalName: "Bonds",
          symbols: [
            {
              name: "CME:GE1!",
              displayName: "Eurodollar"
            },
            {
              name: "CBOT:ZB1!",
              displayName: "T-Bond"
            },
            {
              name: "CBOT:UB1!",
              displayName: "Ultra T-Bond"
            },
            {
              name: "EUREX:FGBL1!",
              displayName: "Euro Bund"
            },
            {
              name: "EUREX:FBTP1!",
              displayName: "Euro BTP"
            },
            {
              name: "EUREX:FGBM1!",
              displayName: "Euro BOBL"
            }
          ]
        },
        {
          name: "Forex",
          originalName: "Forex",
          symbols: [
            {
              name: "FX:EURUSD"
            },
            {
              name: "FX:GBPUSD"
            },
            {
              name: "FX:USDJPY"
            },
            {
              name: "FX:USDCHF"
            },
            {
              name: "FX:AUDUSD"
            },
            {
              name: "FX:USDCAD"
            }
          ]
        }
      ],
      showSymbolLogo: !0,
      colorTheme: "light",
      isTransparent: !1,
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-market-data",
      "tradingview-market-data-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-market-quotes.js"
    );
    return { container: n, tradingview: i };
  }
}), I = ["id"];
function R(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, I);
}
const Oe = /* @__PURE__ */ c(P, [["render", R]]), G = a({
  name: "WMiniChart",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbol: "FX:EURUSD",
      width: 350,
      height: 220,
      locale: "en",
      dateRange: "12M",
      colorTheme: "light",
      trendLineColor: "rgba(41, 98, 255, 1)",
      underLineColor: "rgba(41, 98, 255, 0.3)",
      underLineBottomColor: "rgba(41, 98, 255, 0)",
      isTransparent: !1,
      autosize: !1,
      largeChartUrl: "",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-mini-chart",
      "tradingview-mini-chart-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-mini-symbol-overview.js"
    );
    return { container: n, tradingview: i };
  }
}), W = ["id"];
function Y(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, W);
}
const Ne = /* @__PURE__ */ c(G, [["render", Y]]), V = a({
  name: "WScreener",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      width: 1100,
      height: 523,
      defaultColumn: "overview",
      defaultScreen: "general",
      market: "forex",
      showToolbar: !0,
      colorTheme: "light",
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-screener",
      "tradingview-screener-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-screener.js"
    );
    return { container: n, tradingview: i };
  }
}), H = ["id"];
function Q(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, H);
}
const Me = /* @__PURE__ */ c(V, [["render", Q]]), J = a({
  name: "WSingleTicker",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbol: "FX:EURUSD",
      width: 350,
      colorTheme: "light",
      isTransparent: !1,
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-single-ticker",
      "tradingview-single-ticker-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-single-quote.js"
    );
    return { container: n, tradingview: i };
  }
}), K = ["id"];
function Z(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, K);
}
const Be = /* @__PURE__ */ c(J, [["render", Z]]), z = a({
  name: "WSnaps",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      feedMode: "all_symbols",
      colorTheme: "light",
      isTransparent: !1,
      displayMode: "regular",
      width: 480,
      height: 830,
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-snaps",
      "tradingview-snaps-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-timeline.js"
    );
    return { container: n, tradingview: i };
  }
}), q = ["id"];
function ee(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, q);
}
const Xe = /* @__PURE__ */ c(z, [["render", ee]]), te = a({
  name: "WStockMarket",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      colorTheme: "light",
      dateRange: "12M",
      exchange: "US",
      showChart: !0,
      locale: "en",
      largeChartUrl: "",
      isTransparent: !1,
      showSymbolLogo: !1,
      showFloatingTooltip: !1,
      width: "400",
      height: "600",
      plotLineColorGrowing: "rgba(41, 98, 255, 1)",
      plotLineColorFalling: "rgba(41, 98, 255, 1)",
      gridLineColor: "rgba(240, 243, 250, 0)",
      scaleFontColor: "rgba(120, 123, 134, 1)",
      belowLineFillColorGrowing: "rgba(41, 98, 255, 0.12)",
      belowLineFillColorFalling: "rgba(41, 98, 255, 0.12)",
      belowLineFillColorGrowingBottom: "rgba(41, 98, 255, 0)",
      belowLineFillColorFallingBottom: "rgba(41, 98, 255, 0)",
      symbolActiveColor: "rgba(41, 98, 255, 0.12)",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-stock-market",
      "tradingview-stock-market-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-hotlists.js"
    );
    return { container: n, tradingview: i };
  }
}), ne = ["id"];
function ie(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, ne);
}
const ke = /* @__PURE__ */ c(te, [["render", ie]]);
window.TradingView = window.TradingView || {};
const oe = a({
  name: "WSymbolOverview",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = u("tradingview-symbol-overview"), n = u("tradingview-symbol-overview-script"), i = {
      symbols: [
        ["Apple", "AAPL"],
        ["Google", "GOOGL"],
        ["Microsoft", "MSFT"]
      ],
      chartOnly: !1,
      width: 1e3,
      height: 400,
      locale: "en",
      colorTheme: "light",
      gridLineColor: "rgba(240, 243, 250, 0)",
      fontColor: "#787B86",
      isTransparent: !1,
      autosize: !1,
      showVolume: !1,
      scalePosition: "no",
      scaleMode: "Normal",
      fontFamily: "Trebuchet MS, sans-serif",
      noTimeScale: !1,
      valuesTracking: "1",
      chartType: "area",
      lineColor: "#2962FF",
      bottomColor: "rgba(41, 98, 255, 0)",
      topColor: "rgba(41, 98, 255, 0.3)",
      container_id: "tradingview-symbol-overview"
    }, o = () => typeof window < "u" && window.document && window.document.createElement, r = () => document.getElementById(n.value), g = () => r() !== null, h = (m) => {
      if (!o() || g())
        return;
      const p = document.createElement("script");
      p.id = n.value + Date.now().toString(36), p.type = "text/javascript", p.async = !0, p.src = "https://s3.tradingview.com/tv.js", p.onload = m, document.getElementsByTagName("head")[0].appendChild(p);
    }, f = () => {
      setTimeout(() => {
        if (typeof window.TradingView > "u")
          return;
        const m = { ...i, ...e.options };
        if (m.container_id !== t.value) {
          console.error('container_id in SymbolOverview component must be "tradingview-symbol-overview"');
          return;
        }
        new window.TradingView.MediumWidget(Object.assign({ container_id: t }, m));
      }, 300);
    };
    return v(() => {
      h(() => f());
    }), { container: t };
  }
}), re = ["id"];
function ae(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, re);
}
const Ae = /* @__PURE__ */ c(oe, [["render", ae]]), se = a({
  name: "WTechnicalAnalysis",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      interval: "1m",
      width: 425,
      isTransparent: !1,
      height: 450,
      symbol: "NASDAQ:AAPL",
      showIntervalTabs: !0,
      locale: "en",
      colorTheme: "light",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-technical-analysis",
      "tradingview-technical-analysis-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-technical-analysis.js"
    );
    return { container: n, tradingview: i };
  }
}), de = ["id"];
function ce(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, de);
}
const xe = /* @__PURE__ */ c(se, [["render", ce]]), le = a({
  name: "WTicker",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbols: [
        {
          proName: "FOREXCOM:SPXUSD",
          title: "S&P 500"
        },
        {
          proName: "FOREXCOM:NSXUSD",
          title: "US 100"
        },
        {
          proName: "FX_IDC:EURUSD",
          title: "EUR/USD"
        },
        {
          proName: "BITSTAMP:BTCUSD",
          title: "Bitcoin"
        },
        {
          proName: "BITSTAMP:ETHUSD",
          title: "Ethereum"
        }
      ],
      colorTheme: "light",
      isTransparent: !1,
      showSymbolLogo: !0,
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-ticker",
      "tradingview-ticker-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-tickers.js"
    );
    return { container: n, tradingview: i };
  }
}), pe = ["id"];
function me(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, pe);
}
const je = /* @__PURE__ */ c(le, [["render", me]]), ge = a({
  name: "WTickerTape",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      symbols: [
        {
          proName: "FOREXCOM:SPXUSD",
          title: "S&P 500"
        },
        {
          proName: "FOREXCOM:NSXUSD",
          title: "US 100"
        },
        {
          proName: "FX_IDC:EURUSD",
          title: "EUR/USD"
        },
        {
          proName: "BITSTAMP:BTCUSD",
          title: "Bitcoin"
        },
        {
          proName: "BITSTAMP:ETHUSD",
          title: "Ethereum"
        }
      ],
      showSymbolLogo: !0,
      colorTheme: "light",
      isTransparent: !1,
      displayMode: "adaptive",
      locale: "en",
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-ticker-tape",
      "tradingview-ticker-tape-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-ticker-tape.js"
    );
    return { container: n, tradingview: i };
  }
}), ue = ["id"];
function we(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, ue);
}
const Le = /* @__PURE__ */ c(ge, [["render", we]]), he = a({
  name: "WMarketOverview",
  props: {
    options: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const t = {
      colorTheme: "light",
      dateRange: "12M",
      showChart: !0,
      locale: "en",
      largeChartUrl: "",
      isTransparent: !1,
      showSymbolLogo: !0,
      showFloatingTooltip: !1,
      width: "400",
      height: "660",
      plotLineColorGrowing: "rgba(41, 98, 255, 1)",
      plotLineColorFalling: "rgba(41, 98, 255, 1)",
      gridLineColor: "rgba(240, 243, 250, 0)",
      scaleFontColor: "rgba(120, 123, 134, 1)",
      belowLineFillColorGrowing: "rgba(41, 98, 255, 0.12)",
      belowLineFillColorFalling: "rgba(41, 98, 255, 0.12)",
      belowLineFillColorGrowingBottom: "rgba(41, 98, 255, 0)",
      belowLineFillColorFallingBottom: "rgba(41, 98, 255, 0)",
      symbolActiveColor: "rgba(41, 98, 255, 0.12)",
      tabs: [
        {
          title: "Indices",
          symbols: [
            {
              s: "FOREXCOM:SPXUSD",
              d: "S&P 500"
            },
            {
              s: "FOREXCOM:NSXUSD",
              d: "US 100"
            },
            {
              s: "FOREXCOM:DJI",
              d: "Dow 30"
            },
            {
              s: "INDEX:NKY",
              d: "Nikkei 225"
            },
            {
              s: "INDEX:DEU40",
              d: "DAX Index"
            },
            {
              s: "FOREXCOM:UKXGBP",
              d: "UK 100"
            }
          ],
          originalTitle: "Indices"
        },
        {
          title: "Futures",
          symbols: [
            {
              s: "CME_MINI:ES1!",
              d: "S&P 500"
            },
            {
              s: "CME:6E1!",
              d: "Euro"
            },
            {
              s: "COMEX:GC1!",
              d: "Gold"
            },
            {
              s: "NYMEX:CL1!",
              d: "Crude Oil"
            },
            {
              s: "NYMEX:NG1!",
              d: "Natural Gas"
            },
            {
              s: "CBOT:ZC1!",
              d: "Corn"
            }
          ],
          originalTitle: "Futures"
        },
        {
          title: "Bonds",
          symbols: [
            {
              s: "CME:GE1!",
              d: "Eurodollar"
            },
            {
              s: "CBOT:ZB1!",
              d: "T-Bond"
            },
            {
              s: "CBOT:UB1!",
              d: "Ultra T-Bond"
            },
            {
              s: "EUREX:FGBL1!",
              d: "Euro Bund"
            },
            {
              s: "EUREX:FBTP1!",
              d: "Euro BTP"
            },
            {
              s: "EUREX:FGBM1!",
              d: "Euro BOBL"
            }
          ],
          originalTitle: "Bonds"
        },
        {
          title: "Forex",
          symbols: [
            {
              s: "FX:EURUSD"
            },
            {
              s: "FX:GBPUSD"
            },
            {
              s: "FX:USDJPY"
            },
            {
              s: "FX:USDCHF"
            },
            {
              s: "FX:AUDUSD"
            },
            {
              s: "FX:USDCAD"
            }
          ],
          originalTitle: "Forex"
        }
      ],
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-market-overview",
      "tradingview-market-overview-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-market-overview.js"
    );
    return { container: n, tradingview: i };
  }
}), fe = ["id"];
function ve(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, fe);
}
const Pe = /* @__PURE__ */ c(he, [["render", ve]]), _e = a({
  name: "WSymbolInfo",
  props: {
    options: {
      type: Object,
      default: () => ({
        symbol: "NASDAQ:AAPL",
        width: 1e3,
        locale: "en",
        colorTheme: "light",
        isTransparent: !1
      })
    }
  },
  setup(e) {
    const t = {
      symbol: "NASDAQ:AAPL",
      width: 1e3,
      locale: "en",
      colorTheme: "light",
      isTransparent: !1,
      ...e.options
    }, { container: n, tradingview: i } = l(
      t,
      "tradingview-symbol-info",
      "tradingview-symbol-info-script",
      "https://s3.tradingview.com/external-embedding/embed-widget-symbol-info.js"
    );
    return { container: n, tradingview: i };
  }
}), be = ["id"];
function $e(e, t, n, i, o, r) {
  return s(), d("div", {
    ref: "tradingview",
    id: e.container
  }, null, 8, be);
}
const Ie = /* @__PURE__ */ c(_e, [["render", $e]]);
export {
  Ce as Chart,
  Te as CompanyProfile,
  Se as CryptoMarket,
  Ee as EconomicCalendar,
  Fe as ForexCrossRates,
  Ue as ForexHeatMap,
  De as FundamentalData,
  Oe as MarketData,
  Pe as MarketOverview,
  Ne as MiniChart,
  Me as Screener,
  Be as SingleTicker,
  Xe as Snaps,
  ke as StockMarket,
  Ie as SymbolInfo,
  Ae as SymbolOverview,
  xe as TechnicalAnalysis,
  je as Ticker,
  Le as TickerTape
};
